app/src/main/java/com/trunghoang/aroundhere/data/model/Place.java

```diff
@@ -62,9 +98,32 @@ public Place(JSONObject jsonObject) {
         mAddress = jsonObject.optString(JSONKey.ADDRESS);
         mResId = jsonObject.optString(JSONKey.PLACE_ID);
         mDetailUrl = jsonObject.optString(JSONKey.DETAIL_URL);
+        if (TextUtils.isEmpty(mPhoto)) mPhoto = jsonObject.optString(JSONKey.PHOTO_URL);
+        if (TextUtils.isEmpty(mResId)) mResId = jsonObject.optString(JSONKey.RES_ID);
+        if (TextUtils.isEmpty(mDetailUrl)) mDetailUrl = jsonObject.optString(JSONKey.RES_URL);
+        JSONArray openingTime = jsonObject.optJSONArray(JSONKey.OPENING_TIME);
+        if (openingTime != null && openingTime.length() != 0) {
+            JSONObject time0 = openingTime.optJSONObject(0);
+            JSONObject timeOpen = time0.optJSONObject(JSONKey.TIME_OPEN);
+            JSONObject timeClose = time0.optJSONObject(JSONKey.TIME_CLOSE);
+            mTimeOpen = timeOpen.optLong(JSONKey.TOTAL_MILLS);
+            mTimeClose = timeClose.optLong(JSONKey.TOTAL_MILLS);
+            if (mTimeClose == 0) mTimeClose = FULL_DAY;
+        }
+        mPriceMin = jsonObject.optLong(JSONKey.PRICE_MIN);
+        mPriceMax = jsonObject.optLong(JSONKey.PRICE_MAX);
+        mLat = jsonObject.optDouble(JSONKey.LAT);
+        mLon = jsonObject.optDouble(JSONKey.LON);
     }
 
     public double getDistance() {
```

Comment:
method này lên đựoc tính là calulateDistance thì đúng hơn

Reply:
Method getDistance em dùng trong method showPlace để show dữ liệu mà không cần quan tâm là dữ liệu đó lấy từ local hay remote, mà dữ liệu local thì em không lưu distance vì em muốn mỗi lần lấy ra sẽ tính lại (Đáng ra nếu api hỗ trợ thì em có thể lấy distance mới từ remote nhưng mà nó lại không có ạ hiuhiu). Cho nên em đặt tính lại distance ngay trong getPlace để có thể dùng chung luôn ạ

Reply:
ý anh muốn nói ở đây là method name
https://stackoverflow.com/questions/7151418/determining-which-verb-to-use-for-method-names-in-java
a ko nhớ chính xác nguồn viết nhưng e cũng có thể tham khảo câu hỏi này trên stackoverflow.
Các method trả về các giá trị của field thì e có thể đặt là getXXX
nhưng các method cần tính toán sau đó mới trả về giá trị thì a nghĩ em nên đặt là calucalateXXX

Reply:
để em tách method ạ